src/software/backend/replay_backend.cpp

```diff
@@ -0,0 +1,70 @@
+#include "replay_backend.h"
+
+#include <cstdlib>
+
+#include "software/util/design_patterns/generic_factory.h"
+
+ReplayBackend::ReplayBackend(const std::string& replay_input_dir)
+    : Backend(),
+      replay_reader(replay_input_dir),
+      pull_from_replay_thread(
+          boost::bind(&ReplayBackend::continuouslyPullFromReplayFiles, this)),
+      last_msg_received_time(std::nullopt),
+      last_msg_replayed_time(std::nullopt)
+{
+}
+
+// do nothing
+// TODO: #1745 perhaps we want to encode the output primitives to proto and log them?
```

Comment:
Putting the issue link as well would be nice to keep it consistent with the rest of the TODOs.

Reply:
issue link is not required; either way is fine